public/video-ui/src/services/VideosApi.js

```diff
@@ -86,55 +86,54 @@ export default {
     });
   },
 
-  updateComposerPage(id, metadata, composerUrlBase, videoBlock, usages) {
+  updateComposerPage(id, video, composerUrlBase, videoBlock, usages) {
     function getComposerUpdateRequests(
       id,
-      metadata,
+      video,
       composerUrlBase,
       videoBlock,
       isLive
     ) {
+      const rightsExpiryPayload = getRightsPayload(video);
+      const rightsRequest = pandaReqwest({
+        url: `${composerUrlBase}/api/content/${id}/expiry/rights`,
+        method: 'post',
+        contentType: 'application/json',
+        crossOrigin: true,
+        withCredentials: true,
+        data: JSON.stringify(rightsExpiryPayload)
+      });
+
       // When article is in preview, composer keeps track of both the live and preview versions of an article
       // For an uplublished article, we need to update both live and preview versions.
-      return composerSyncFields.reduce((promises, field) => {
-        promises.push(
-          updateArticleField(
-            'preview',
-            field,
-            metadata[field],
-            composerUrlBase,
-            id
-          )
-        );
+      const composerData = getComposerData(video);
+      const dataUpdatePromises = composerData.reduce((promises, data) => {
+        promises.push(updateArticleField('preview', data, composerUrlBase, id));
+
         if (!isLive) {
-          promises.push(
-            updateArticleField(
-              'live',
-              field,
-              metadata[field],
-              composerUrlBase,
-              id
-            )
-          );
+          promises.push(updateArticleField('live', data, composerUrlBase, id));
         }
         return promises;
       }, []);
+
+      dataUpdatePromises.push(rightsRequest);
+      return dataUpdatePromises;
     }
 
-    function updateArticleField(stage, field, value, composerUrl, pageId) {
-      if (value) {
+    function updateArticleField(stage, data, composerUrl, pageId) {
+      if (data.value || data.belongsTo === 'settings') {
         return pandaReqwest({
-          url: `${composerUrl}/api/content/${pageId}/${stage}/fields/${field}`,
+          url: `${composerUrl}/api/content/${pageId}/${stage}/${data.belongsTo}/${data.name}`,
           method: 'put',
           contentType: 'application/json',
           crossOrigin: true,
           withCredentials: true,
-          data: value
+          data: JSON.stringify(data.value)
```

Comment:
Since #424 we don't need to `JSON.stringify` `data` here. `contentType` is also [defaulted](https://github.com/guardian/media-atom-maker/blob/master/public/video-ui/src/services/pandaReqwest.js#L40-L47) to `application/json` now too.